tools/windows/DatadogAgentInstaller/CustomActions/UserCustomActions.cs

```diff
@@ -437,5 +438,71 @@ public static ActionResult ConfigureUser(Session session)
         {
             return ConfigureUser(new SessionWrapper(session));
         }
+
+        private static ActionResult OpenMsiLog(ISession session)
+        {
+            // The MessageBoxs are unlikely to ever show.
+            // In testing I couldn't hit the "failed" ones. For permissions errors,
+            // Notepad still opens, and Notepad shows its own error dialog box.
+            // The log file should always exist because we set the MsiLogging
+            // property, so unless that changes we won't see the "no log file"
+            // MessageBoxs either.
+            // The log file can't be deleted/renamed while the installer is running
+            // because the installer has a handle to it.
+            // We use MessageBoxIcon.Warning rather than MessageBoxIcon.Error
+            // to match the WiX built-in error dialogs.
+            var wixLogLocation = string.Empty;
+            var messageBoxTitle = "Datadog Agent Setup";
+            try
+            {
+                wixLogLocation = session["MsiLogFileLocation"];
+                if (!string.IsNullOrEmpty(wixLogLocation))
+                {
+                    var proc = System.Diagnostics.Process.Start(wixLogLocation);
```

Comment:
that's a really hard question to answer.

[VirusTotal](https://www.virustotal.com/gui/file/083267c833d8cd13368e3689a3b5169bc933dc28d03d7f882f054d02984b8de9) is okay with it.

The code only executes when the user clicks the button, so that particular code path probably wasn't dynamically tested/executed by VirusTotal. But at least the AVs are okay with it statically having the ability to execute another process.